src/Data/Map/Monoidal.hs

```diff
@@ -216,6 +221,10 @@ instance (Ord k, Semigroup a) => Monoid (MonoidalMap k a) where
     {-# INLINE mappend #-}
 #endif
 
+instance Ord k => Alt (MonoidalMap k) where
```

Comment:
GHC seems to get confused when trying to derive this one due to the `Applicative` constraint in `some/many`.

with `deriving Alt` I get 
```
src/Data/Map/Monoidal.hs:167:15-17: error:
    • Couldn't match type ‘M.Map k’ with ‘MonoidalMap k’
        arising from the coercion of the method ‘many’
          from type ‘forall a.
                     Applicative (M.Map k) =>
                     M.Map k a -> M.Map k [a]’
            to type ‘forall a.
                     Applicative (MonoidalMap k) =>
                     MonoidalMap k a -> MonoidalMap k [a]’
    • When deriving the instance for (Alt (MonoidalMap k))
    |
167 |               Alt, Apply, Bind)
```

while `deriving instance (Ord k) => Alt (MonoidalMap k)` gives me
```
src/Data/Map/Monoidal.hs:169:1-48: error:
    • Could not deduce (Applicative (M.Map k))
        arising from a use of ‘some’
      from the context: Ord k
        bound by the instance declaration
        at src/Data/Map/Monoidal.hs:169:1-48
      or from: Applicative (MonoidalMap k)
        bound by the type signature for:
```